client/src/main/scala/io/delta/sharing/client/DeltaSharingClient.scala

```diff
@@ -1023,7 +1033,7 @@ class DeltaSharingRestClient(
           var additionalErrorInfo = ""
           if (statusCode == HttpStatus.SC_UNAUTHORIZED && tokenExpired(profile)) {
             additionalErrorInfo = s"It may be caused by an expired token as it has expired " +
-              s"at ${profile.expirationTime}"
+              s"at ${profile.asInstanceOf[BearerTokenDeltaSharingProfile].expirationTime}"
```

Comment:
why is it safe to call asInstanceOf here?

Reply:
did abstraction using the authCredentailProvider.